include/universal/utility/evaluate_closure_of_number_systems.hpp

```diff
@@ -0,0 +1,202 @@
+#pragma once
+
+// Copyright (C) 2017 Stillwater Supercomputing, Inc.
+// SPDX-License-Identifier: MIT
+//
+// Author: Colby Wirth
+// Version: 18 April 2025
+// 
+// About:
+// This utility calls the buildClosurePlot function from genereateClosurePlots.hpp 
+// and generates the closure statistics for a templatized number system. 
+//
+//
+// ****************************************************************
+// *** Use 'processASystem()' function call to use this utility ***
+// ****************************************************************
+// 
+// This file is part of the universal numbers project, which is released under an MIT Open Source license.
+
+#include <iostream>
+#include <string>
+#include <string_view>
+#include <fstream>
+
+#include <universal/number/posit/posit.hpp>
+#include <universal/number/cfloat/cfloat.hpp>
+#include <universal/number/lns/lns.hpp>
+#include <universal/utility/generateClosurePlots.hpp>
+
+
+// the data for each operation in a buildClosurePlots is stored in this struct
+struct operationResults{
+    float TotalOps;
+    std::string Exact;
+    std::string Approximate;
+    std::string Overflow;
+    std::string Underflow;
+    std::string Saturate;
+    std::string narOrNan;
```

Comment:
narOrNan has a different camelcase form than the other vars, which jumps out at me. Secondly, the name narOrNan is clearly an 'unclean' concept, even the name can't figure out what it stands for. This is an architectural design decision, so needs careful analysis as it needs to be unambiguous in its purpose.

Personally, Not A Real, is not as widespread in use as Not A Number. John's argument is that NaN is not kosher because the value is being used as a number. And his improvement to Not A Real was to resolve that tension. Problem is that it has not been picked up, and there is more familiarity with NaN.

You can make the call what you would like to call it, but make it ONE concept (so remove the or) and name it with the same CamelCase of the other vars.